src/porepy/numerics/interface_laws/contact_mechanics_interface_laws.py

```diff
@@ -0,0 +1,781 @@
+        #
+        # These are the contact conditions, which dictate relations between
+        # the contact forces on the slave, and the displacement jumps.
+        #
+        # NOTE: Both the contact conditions and the contact stresses are defined in the
+        # local coordinate system of the surface. The displacements must therefore
+        # be rotated to this local coordinate system during assembly.
+        traction_discr, displacement_jump_discr, rhs_slave = self.discr_slave.assemble_matrix_rhs(
+            g_slave, data_slave
+        )
+        # The contact forces. Can be applied directly, these are in their own
+        # local coordinate systems.
+        cc[slave_ind, slave_ind] = traction_discr
+
+        # The contact condition discretization gives coefficients for the mortar
+        # variables. To finalize the relation with the contact conditions, we
+        # (from the right) 1) assign +- signs to the two sides of the mortar, so that
+        # summation in reality is a difference, 2) project to the mortar grid
+        # 3) project to the local coordinates of the fracture, 4) assign the
+        # coefficients of the displacement jump.
+        cc[slave_ind, mortar_ind] = (
+            displacement_jump_discr
+            * projection.project_tangential_normal(g_slave.num_cells)
+            * mg.mortar_to_slave_avg(nd=ambient_dimension)
+            * mg.sign_of_mortar_sides(nd=ambient_dimension)
+        )
+
+        # Right hand side system. In the local (surface) coordinate system.
+        rhs[slave_ind] = rhs_slave
+
+        ### Equation for the mortar rows
+
+        # This is first a stress balance: stress from the higher dimensional
+        # domain (both interior and bound_stress) should match with the contact stress:
+        # -\lambda_slave + \lambda_mortar = 0.
+        # Optionally, a diffusion term can be added in the tangential direction
+        # of the stresses, this is currently under implementation.
+
+        # A diagonal operator is needed to switch the sign of vectors on
+        # higher-dimensional faces that point into the fracture surface. The effect is to
+        # switch direction of the stress on boundary for the higher dimensional domain: The
+        # contact forces are defined as negative in contact, whereas the sign of the higher
+        # dimensional stresses are defined according to the direction of the normal vector.
+        faces_on_fracture_surface = mg.master_to_mortar_int().tocsr().indices
+        sign_switcher = pp.grid_utils.switch_sign_if_inwards_normal(
+            g_master, ambient_dimension, faces_on_fracture_surface
+        )
+
+        ## First, we obtain \lambda_mortar = stress * u_master + bound_stress * u_mortar
+        # Stress contribution from the higher dimensional domain, projected onto
+        # the mortar grid
+        # Switch the direction of the vectors, so that for all faces, a positive
+        # force points into the fracture surface.
```

Comment:
"...into the fracture surface" -> from the rock matrix to the fracture 
Or rewrite comment:     
   Switch the direction of the vectors, to obtain the traction as defined from the outwards pointing normal vector.

Reply:
yes.